### PR TITLE
Develop #19 implement event category

### LINKED ISSUE #19: [T] Implement Event-Category
# Description
This task will create the necessary components for the Event-category feature.

# Acceptance criteria
- Proper functionality of the Event-category feature
- DTOs for Create, Delete, and Update are working correctly
- Database entity with Liquibase is created and properly linked
- Handler/service is properly implemented
- Endpoint/controller is properly implemented
- Validators are working correctly

# Dependencies
<List any tasks that must be completed before or after this task>
<Include links to related tasks or documentation>

### PR DESCRIPTION
**Pull request by jaegerdominik**
# 💻 What
TODO

# ❓ Why
TODO

# 📘 How
TODO

# 👀 See
Screenshots

### CODE REVIEW

File: src/main/java/at/fh/joanneum/irfc/rest/EventCategoryApi.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  public List<EventCategoryDTO> getAll(){
+    return eventCategoryService.getAll();
+  }
+
+  @POST
+  @Produces(MediaType.APPLICATION_JSON)
+  @Consumes(MediaType.APPLICATION_JSON)
+  public EventCategoryDTO create(EventCategoryDTO eventCategoryDTOCreate){
```
Comment (LionKornsteiner) [inline src/main/java/at/fh/joanneum/irfc/rest/EventCategoryApi.java]: Rename eventCategoryDtoCreate to eventCategoryDto

File: src/main/java/at/fh/joanneum/irfc/rest/EventCategoryApi.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  @Path("/{id}")
+  public EventCategoryDTO getById(@PathParam("id") Long id){
+    return eventCategoryService.get(id);
+  }
+
+  @PUT
+  @Produces(MediaType.APPLICATION_JSON)
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Path("/{id}")
+  public EventCategoryDTO update(@PathParam("id") Long id, EventCategoryDTO eventCategoryDTOUpdate){
```
Comment (LionKornsteiner) [inline src/main/java/at/fh/joanneum/irfc/rest/EventCategoryApi.java]: Rename eventCategoryDtoUpdate to eventCategoryDto

File: src/main/java/at/fh/joanneum/irfc/service/EventCategoryService.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  EventCategoryRepository eventCategoryRepository;
+  public List<EventCategoryDTO> getAll() {
+    return eventCategoryRepository.listAll().stream()
+        .map(EventCategoryMapper.INSTANCE::toDto)
+        .collect(Collectors.toUnmodifiableList());
+  }
+
+  @Transactional
+  public EventCategoryDTO create(EventCategoryDTO eventCategoryDTOCreate) {
+    EventCategoryEntity newEntity = new EventCategoryEntity();
```
Comment (LionKornsteiner) [inline src/main/java/at/fh/joanneum/irfc/service/EventCategoryService.java:46]: Add validator, see pull request of moe

File: src/main/java/at/fh/joanneum/irfc/service/EventCategoryService.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  @Inject
+  EventCategoryRepository eventCategoryRepository;
+  public List<EventCategoryDTO> getAll() {
+    return eventCategoryRepository.listAll().stream()
+        .map(EventCategoryMapper.INSTANCE::toDto)
+        .collect(Collectors.toUnmodifiableList());
+  }
+
+  @Transactional
+  public EventCategoryDTO create(EventCategoryDTO eventCategoryDTOCreate) {
```
Comment (LionKornsteiner) [inline src/main/java/at/fh/joanneum/irfc/service/EventCategoryService.java]: Rename do like above

File: src/main/java/at/fh/joanneum/irfc/service/EventCategoryService.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    EventCategoryEntity newEntity = new EventCategoryEntity();
+    setValues(eventCategoryDTOCreate, newEntity);
+    eventCategoryRepository.persist(newEntity);
+    return EventCategoryMapper.INSTANCE.toDto(newEntity);
+  }
+
+  @Transactional
+  public EventCategoryDTO update(Long id, EventCategoryDTO eventCategoryDTOUpdate) {
```
Comment (LionKornsteiner) [inline src/main/java/at/fh/joanneum/irfc/service/EventCategoryService.java]: Rename do like above and add validator , see pull request of moe

File: src/main/resources/db/19-create-event-category-table.xml [hunk 6]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
```
Comment (LionKornsteiner) [inline src/main/resources/db/19-create-event-category-table.xml:1]: Add migration to changelog.xml

File: src/main/java/at/fh/joanneum/irfc/model/eventCategory/EventCategoryDTO.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * @author dominik.jaeger@edu.fh-joanneum.at
+ **/
+@NoArgsConstructor
+@Getter
+@Setter
+@RegisterForReflection
+public class EventCategoryDTO {
+
+  private Long eventCategoryId;
```
Comment (LionKornsteiner) [inline src/main/java/at/fh/joanneum/irfc/model/eventCategory/EventCategoryDTO.java:17]: why is the eventCategoryId private?

File: src/main/java/at/fh/joanneum/irfc/service/EventCategoryService.java [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    } else {
+      EventCategoryEntity byId = byIdOptional.get();
+      return EventCategoryMapper.INSTANCE.toDto(byId);
+    }
+  }
+
+  @Transactional
+  public EventCategoryDTO create(EventCategoryDTO eventCategoryDTO) {
+
+    checkDTOvalues(eventCategoryDTO);
```
Comment (LionKornsteiner) [inline src/main/java/at/fh/joanneum/irfc/service/EventCategoryService.java]: maybe rename it to validateDtoValues or something like that

**Review by LionKornsteiner — APPROVED** [review]

File: src/main/java/at/fh/joanneum/irfc/model/eventCategory/EventCategoryDTO.java (hunk 7 above)
Comment (LionKornsteiner) [reply src/main/java/at/fh/joanneum/irfc/model/eventCategory/EventCategoryDTO.java:17]: @PhilipMoser why do we make the Id in the Dto private?
@jaegerdominik sry i just checked in the old PR, please make the id private again, even tough im not sure why

Comment (PhilipMoser) [reply src/main/java/at/fh/joanneum/irfc/model/eventCategory/EventCategoryDTO.java:17]: @LionKornsteiner why would you make it public? you can interact with the id over setters and getters. the real question is: why are not all members private, or i'm wrong?

Comment (LionKornsteiner) [reply src/main/java/at/fh/joanneum/irfc/model/eventCategory/EventCategoryDTO.java:17]: @PhilipMoser @jaegerdominik makes, sense, dominik, pls make all members private

File: src/main/java/at/fh/joanneum/irfc/service/EventCategoryService.java [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+  @Inject
+  EventCategoryRepository eventCategoryRepository;
+  public List<EventCategoryDTO> getAll() {
+    return eventCategoryRepository.listAll().stream()
+        .map(EventCategoryMapper.INSTANCE::toDto)
+        .collect(Collectors.toUnmodifiableList());
+  }
+
+  public EventCategoryDTO get(Long id) {
```
Comment (marioradman) [inline src/main/java/at/fh/joanneum/irfc/service/EventCategoryService.java:31]: Warum verwendest du hier den Long Wrapper?

**Review by marioradman — APPROVED** [review]
LGTM, nitpick

File: src/main/java/at/fh/joanneum/irfc/service/EventCategoryService.java (hunk 9 above)
Comment (jaegerdominik) [reply src/main/java/at/fh/joanneum/irfc/service/EventCategoryService.java:31]: cause the id is a Long in our defined data model

File: src/main/java/at/fh/joanneum/irfc/model/eventCategory/EventCategoryDTO.java (hunk 7 above)
Comment (jaegerdominik) [reply src/main/java/at/fh/joanneum/irfc/model/eventCategory/EventCategoryDTO.java:17]: done

**Review by GoldNova — COMMENTED** [review]
LGTM

**Review by GoldNova — APPROVED** [review]
LGTM

**Review by GoldNova — APPROVED** [review]
LGTM

**Pull request merged**